### ISSUE TITLE
mitigating ratelimits while setting up environment

### ISSUE THREAD

**Issue by AskAlice** [arpanghosh8453/public-fitbit-projects#18]
I think it should first try to connect to influxdb before making any fitbit api calls. No reason to collect the data if it can't report it.

**Comment by arpanghosh8453** [comment]
Hey @AskAlice, Your point is valid. The container is designed to pull data on a rolling basis, so if you fix the connection, the next time, it will write properly. Thanks for taking a closer look to the codebase and pointing this out. Perks of open source project I guess. I agree on the fact that there should be a initial check which confirms a valid influxdb connection before starting everything. That will be an easy fix!

**Comment by arpanghosh8453** [comment]
I think this is already covered by the following code - (line #169 - #182)
```python
if INFLUXDB_VERSION == "2":
    try:
        influxdbclient = InfluxDBClient2(url=INFLUXDB_URL, token=INFLUXDB_TOKEN, org=INFLUXDB_ORG)
        influxdb_write_api = influxdbclient.write_api(write_options=SYNCHRONOUS)
    except InfluxDBError as err:
        logging.error("Unable to connect with influxdb 2.x database! Aborted")
        raise InfluxDBError("InfluxDB connection failed:" + str(err))
elif INFLUXDB_VERSION == "1":
    try:
        influxdbclient = InfluxDBClient(host=INFLUXDB_HOST, port=INFLUXDB_PORT, username=INFLUXDB_USERNAME, password=INFLUXDB_PASSWORD)
        influxdbclient.switch_database(INFLUXDB_DATABASE)
    except InfluxDBClientError as err:
        logging.error("Unable to connect with influxdb 1.x database! Aborted")
        raise InfluxDBClientError("InfluxDB connection failed:" + str(err))
```

I can't test for influxdb 2.0+ but I think influxdb 1.8 will get terminated before the fetching starts if the connection fails. This check is done only when the script starts. if not, this can be injected in that place!

Influxdb 2.0+ provides a way to [test the connection](https://github.com/influxdata/influxdb-client-python/issues/131) like this :

```python
from influxdb_client import InfluxDBClient

client = InfluxDBClient(url="http://localhost:8080",
                        token="my-token",
                        org="my-org",)

health = client.health()
if health.status == "pass":
    print("Connection success.")
else:
    print(f"Connection failure: {health.message}!")

client.__del__()
```
But I am not sure what's the influxdb 1.0 equivalent (or even this health check feature is available or not)

**Comment by AskAlice** [comment]
> I think this is already covered by the following code - (line #169 - #182)
> 
> ```python
> if INFLUXDB_VERSION == "2":
>     try:
>         influxdbclient = InfluxDBClient2(url=INFLUXDB_URL, token=INFLUXDB_TOKEN, org=INFLUXDB_ORG)
>         influxdb_write_api = influxdbclient.write_api(write_options=SYNCHRONOUS)
>     except InfluxDBError as err:
>         logging.error("Unable to connect with influxdb 2.x database! Aborted")
>         raise InfluxDBError("InfluxDB connection failed:" + str(err))
> elif INFLUXDB_VERSION == "1":
>     try:
>         influxdbclient = InfluxDBClient(host=INFLUXDB_HOST, port=INFLUXDB_PORT, username=INFLUXDB_USERNAME, password=INFLUXDB_PASSWORD)
>         influxdbclient.switch_database(INFLUXDB_DATABASE)
>     except InfluxDBClientError as err:
>         logging.error("Unable to connect with influxdb 1.x database! Aborted")
>         raise InfluxDBClientError("InfluxDB connection failed:" + str(err))
> ```
> 
> I can't test for influxdb 2.0+ but I think influxdb 1.8 will get terminated before the fetching starts if the connection fails. This check is done only when the script starts. if not, this can be injected in that place!
> 
> Influxdb 2.0+ provides a way to [test the connection](https://github.com/influxdata/influxdb-client-python/issues/131) like this :
> 
> ```python
> from influxdb_client import InfluxDBClient
> 
> client = InfluxDBClient(url="http://localhost:8080",
>                         token="my-token",
>                         org="my-org",)
> 
> health = client.health()
> if health.status == "pass":
>     print("Connection success.")
> else:
>     print(f"Connection failure: {health.message}!")
> 
> client.__del__()
> ```
> 
> But I am not sure what's the influxdb 1.0 equivalent (or even this health check feature is available or not)

at the very least It depends on timezone being set to something other than automatic
